airflow/providers/amazon/aws/sensors/dynamodb.py

```diff
@@ -0,0 +1,96 @@
+# under the License.
+from __future__ import annotations
+
+import os
+import boto3
+from pprint import pprint
+from botocore.exceptions import ClientError
+
+from airflow.sensors.base import BaseSensorOperator
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class DynamoDBSensor(BaseSensorOperator):
+    """
+    Waits for an attribute value to be present for an item in a DynamoDB table.
+
+    :param partition_key_name: DynamoDB partition key name
+    :param partition_key_value: DynamoDB partition key value
+    :param attribute_name: DynamoDB attribute name
+    :param attribute_value: DynamoDB attribute value
+    :param sort_key_name: (optional) DynamoDB sort key name
+    :param sort_key_value: (optional) DynamoDB sort key value
+    """
+
+    def __init__(
+        self,
+        table_name: str,
+        partition_key_name=None,
+        partition_key_value=None,
+        attribute_name=None,
+        attribute_value=None,
+        sort_key_name: Optional[str] = None,
+        sort_key_value: Optional[str] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.partition_key_name = partition_key_name
+        self.partition_key_value = partition_key_value
+        self.attribute_name = attribute_name
+        self.attribute_value = attribute_value
+        self.sort_key_name = sort_key_name
+        self.sort_key_value = sort_key_value
+        self.dynamodb = boto3.resource("dynamodb", region_name=os.getenv("AWS_REGION"))
+        self.table_name = table_name
+        self.table = self.dynamodb.Table(table_name)
+
+    def poke(self, context: Context) -> bool:
+        """Test DynamoDB item for matching attribute value"""
+        try:
+            if self.sort_key_name and self.sort_key_value:
+                print(
+                    f"Checking table {self.table_name} for item PK: {self.partition_key_name}={self.partition_key_value} SK: {self.sort_key_name}={self.sort_key_value} attribute: {self.attribute_name}={self.attribute_value}"
+                )
+                response = self.table.get_item(
+                    Key={
+                        self.partition_key_name: self.partition_key_value,
+                        self.sort_key_name: self.sort_key_value,
+                    }
+                )
+            else:
+                print(
+                    f"Checking table {self.table_name} for item PK: {self.partition_key_name}={self.partition_key_value} attribute: {self.attribute_name}={self.attribute_value}"
+                )
+                response = self.table.get_item(Key={self.partition_key_name: self.partition_key_value})
+
+            if not response:
+                return False
+        except ClientError as error:
+            if error.response["Error"]["Code"] == "ResourceNotFoundException":
+                print("Table not found: " + self.table_name)
+                return False
```

Comment:
does it make sense to continue waiting until the table is created ? Would that be a use case ?
just asking to make sure I understand the reasoning to isolate that case (maybe worth a comment explaining why it's treated differently)

Reply:
I'm isolating it here for the sole purpose of saying "Table not found" to make it obvious in the logs. Waiting for table creation is not the intent here. Presumably, the table will exist long before this sensor is invoked.

Reply:
Changed to:
```python
 except ClientError as ex:
    self.log.error('AWS request failed, check logs for more info: %s', ex)
    return False
 ```